wayland-protocols/ext-lockscreen_v1.xml

```diff
@@ -0,0 +1,104 @@
+    under the terms of the GNU General Public License version 2 or 3,
+    as published by the Free Software Foundation.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  </copyright>
+  <description summary="compositor authoritative screen-locking mechanism">
+    This protocol allows for a priveleged Wayland client to display arbitrary graphics
+    on the screen when the compositor informs the client that the session has been
+    locked.
+
+    The compositor may choose to restrict this protocol to a special client
+    launched by the compositor itself or expose it to all privileged clients.
+    This is up to the compositor policy.
+
+    The compositor is responsible for deciding when the session should be locked and unlocked.
+    If the client dies while the session is locked, the client may simply resubmit
+    a buffer for display at a later time. During this time, the compositor itself
+    may have been unlocked via some other mechanism (e.g. loginctl). Exactly how and
+    when the compositor locks and unlocks is the policy of the compositor.
+
+    A typical use case is for the compositor to lock or unlock when it receives a
+    signal from Logind to do so.
+
+    The key words "must", "must not", "required", "shall", "shall not",
+    "should", "should not", "recommended",  "may", and "optional" in this
+    document are to be interpreted as described in IETF RFC 2119.
+  </description>
+
+  <interface name="ext_lockscreen_manager_v1" version="1">
+    <description summary="used to create lockscreen surfaces">
+      
+    </description>
+    <request name="associate">
+      This request creates a lockscreen surface and asks the compositor to show this
+      surface on the provided output when the session is next locked. If the session is
+      already locked, the compositor will display this surface right away.
+
+      Surfaces may display different content on different outputs.
+      
+      This surface will be shown immediately after the session is locked. It is expected
+      that the compositor does not render intermediary content if a lockscreen surface
+      is provided for an output. If a surface is NOT provided for an output, then it is up to the
+      policy of the compositor to render what it wants.
+
+      The client is expected to destroy lockscreen surfaces as outputs are removed.
+      The client is expected to associate lockscreen surfaces as outputs are added.
+
+      Associating more than one surface for an output is a duplicate_surface error.
+      Associating a surface with a null buffer is a null_buffer error.
```

Comment:
What happens if a client waits until receiving the `locked` event to commit its first buffer?

Reply:
Good question. I am assuming a protocol error